src/test/java/com/marklogic/kafka/connect/BuildDatabaseClientConfigTest.java

```diff
@@ -160,21 +161,122 @@ public void gatewayConnection() {
     }
 
     @Test
-    public void testInvalidAuthentication() {
-        Map<String, Object> config = new HashMap<>();
-        config.put(MarkLogicSinkConfig.CONNECTION_HOST, "");
-        config.put(MarkLogicSinkConfig.CONNECTION_PORT, 8000);
-        config.put(MarkLogicSinkConfig.CONNECTION_SECURITY_CONTEXT_TYPE, "IncorrectValue");
-        ConfigException ex = assertThrows(ConfigException.class, () -> MarkLogicSinkConfig.CONFIG_DEF.parse(config),
+    public void testInvalidConfiguration() {
```

Comment:
This looks like it should be broken up into several smaller tests. The test originally had a scope of verifying that invalid authentication was handled. The new test is much more difficult to understand since there are so many lines of code in it. Think it'd be much better and easier to understand if there's a test per config option - e.g. testInvalidDocumentFormat.